docs/scripts/render_hubs.py

```diff
@@ -98,59 +61,195 @@
     elif provider == "azure":
         cluster_console_url = "https://portal.azure.com"
 
-    # For each hub in cluster, grab its metadata and add it to the list
-    for hub in cluster["hubs"]:
-        # Domain can be a list
-        if isinstance(hub["domain"], list):
-            hub["domain"] = hub["domain"][0]
-
-        hub_list.append(
-            {
-                # Fallback to name if display_name isn't available
-                "name": hub.get("display_name", "name"),
-                "domain": f"[{hub['domain']}](https://{hub['domain']})",
-                "id": hub["name"],
-                "hub_type": hub["helm_chart"],
-                "grafana": grafana_url,
-                "cluster": cluster["name"],
-                "provider": provider,
-                "data center location": datacentre_loc,  # Americanising for you ;)
-                "UI console link": f"[Use with **{account}** account]({cluster_console_url})",
-                "admin_url": f"[admin](https://{hub['domain']}/hub/admin)"
-                if cluster_console_url
-                else None,
-            }
+    return cluster_console_url
+
+
+def get_hub_authentication(cluster_path, hub_values_files):
+    # Loop through hub values files, look for authenticator_class config, and grab the info
+    authenticator_info = ""
+    for config_file in hub_values_files:
```

Comment:
I wonder if there's a way to write this loop such that if we populate `authenticator_info` on the pass of the first file, we can skip reading in the following files?

Reply:
Yes, will do that in an upcoming commit. But I believe we will end up restructuring this function anyway really soon. This is because most of the other features will require searching these config values files and we really want to open all of them once and get all the info we want about the hub options.